crates/trace-data/src/lib.rs

```diff
@@ -11,22 +11,24 @@ pub struct ContractAddress(pub String);
 #[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
 pub struct EntryPointSelector(pub String);
 
+/// Enum representing node of a trace of a call.
+/// A trace root should always be of variant [`CallTraceNode::Regular`].
```

Comment:
This could be probably enforcable on the type level, can we create a special root type or just require a `CallTrace`?

Reply:
Requiring `CallTrace` seems fine